idaes/generic_models/properties/core/generic/generic_property.py

```diff
@@ -1393,6 +1416,77 @@ def keq_rule(b, r):
                 doc="Inherent reaction equilibrium constraint",
                 rule=equil_rule)
 
+    def calculate_scaling_factors(self):
+        # Get default scale factors and do calculations from base classes
+        super().calculate_scaling_factors()
+
+        sf_T = iscale.get_scaling_factor(
+            self.temperature, default=1, warning=True)
+        sf_P = iscale.get_scaling_factor(
+            self.pressure, default=1, warning=True)
+        sf_mf = iscale.get_scaling_factor(
```

Comment:
This uses the `get_scaling_factors` method - it will look first to the default values defined for the parameter block before falling back to the hard-coded default value. Thus, these sorts of things should be defined in the parameter block, not here.

Note also that calculating these sorts of things from the flowrate is challenging. Firstly - how do you get good values for the flowrates to begin with (especially in a generic sense where they may not be state variables themselves). Secondly, what if the current value happens to be badly off?